### PR TITLE
UWP work for System.Net.NetworkInformation

### PR DESCRIPTION
**Pull request by schen381**
1. Disabled some tests for UAP/ILC runs
2. Remove dependency on Win32 APIs

System.Net.NetworkInformation now builds clean and has no test failures on uap & uapaot

contributes to: #20014

### CODE REVIEW

**Comment by schen381** [comment]
I found that a Win32 api `SetFileCompletionNotificationModes` is not supported on UAP. Inside `SafeCloseSocket.Windows.cs`, there is a function `GetOrAllocateThreadPoolBoundHandle()` will eventually called into that api. Even if we never use `GetOrAllocateThreadPoolBoundHandle()` in S.N.NetworkInformation, the compiler will check the entire context for `SafeCloseSocket.Windows.cs`. This file has many dependencies and many other files depend on it as well, therefore it's hard to merge (windows & !uap) and (windows & uap) clean in .csproj. My approach is to match UAP's behavior and delete `CompletionPortHelper.SkipCompletionPortOnSuccess()` in `SafeCloseSocket.Windows.cs` (in UAP `SkipCompletionPortOnSuccess` will always return false). This apparently is not good, and any suggestions on this will be appreciated! I will mark this PR as no merge for now.

File: src/Common/src/System/Net/SafeCloseSocket.Windows.cs [hunk 1]
```diff
@@ -64,13 +64,8 @@ public ThreadPoolBoundHandle GetOrAllocateThreadPoolBoundHandle(bool trySkipComp
...
                         }
-
-                        // Try to disable completions for synchronous success, if requested
-                        if (trySkipCompletionPortOnSuccess &&
-                            CompletionPortHelper.SkipCompletionPortOnSuccess(boundHandle.Handle))
-                        {
-                            _skipCompletionPortOnSuccess = true;
-                        }
+                        
+                        _skipCompletionPortOnSuccess = false;
```
Comment (stephentoub) [inline src/Common/src/System/Net/SafeCloseSocket.Windows.cs]: Why did you change this?  Wasn't there an issue on Win7 / UDP with this?

Comment (schen381) [inline src/Common/src/System/Net/SafeCloseSocket.Windows.cs]: please see my comment here: https://github.com/dotnet/corefx/pull/20650#issuecomment-305923505

Comment (stephentoub) [inline src/Common/src/System/Net/SafeCloseSocket.Windows.cs]: We must not change this.  You will need to find another workaround.

Comment (CIPop) [inline src/Common/src/System/Net/SafeCloseSocket.Windows.cs]: We shouldn't need any changes to production code.

@schen381 let's take this offline as it requires more investigation regarding missing APIs.

**Comment by schen381** [comment]
After offline discussion, I need to close this PR and delete one of my comment.

File: src/Common/src/System/Net/SafeCloseSocket.Windows.cs (hunk 1 above)
Comment (stephentoub) [inline src/Common/src/System/Net/SafeCloseSocket.Windows.cs]: Thanks
